### PR TITLE
Update Wrapper for Beacon State

### PR DESCRIPTION
**Pull request by nisdas**
This adds in missing fields and getter and setter methods for the beacon state.

### CODE REVIEW

File: beacon-chain/core/types/state.go [hunk 1]
```diff
@@ -192,9 +224,60 @@ func (b *BeaconState) LatestCrosslinks() []*pb.CrosslinkRecord {
...
-func (b *BeaconState) ValidatorRegistry() []*pb.ValidatorRecord {
-	return b.data.ValidatorRegistry
+// NextSeedHash returns the next seed to be used in RANDAO.
+func (b *BeaconState) NextSeedHash() [32]byte {
+	var h [32]byte
+	copy(h[:], b.data.NextSeedHash32)
+	return h
+}
+
+// PersistentCommittees returns the stored in the beacon state.
```
Comment (prestonvanloon) [inline beacon-chain/core/types/state.go]: Missing word

File: beacon-chain/core/types/state.go [hunk 2]
```diff
@@ -395,6 +478,76 @@ func (b *BeaconState) SetValidatorRegistryLastChangeSlot(slot uint64) {
...
+func (b *BeaconState) SetJustifiedSlotBitfield(field uint64) {
+	b.data.JustifiedSlotBitfield = field
+}
+
+// SetLatestAttestations sets the latest pending attestations into the state.
+func (b *BeaconState) SetLatestAttestations(attestations []*pb.PendingAttestationRecord) {
+	b.data.LatestAttestations = attestations
+}
+
+// SetProcessedPowReceiptHash saves the POW recipts which have
```
Comment (prestonvanloon) [inline beacon-chain/core/types/state.go]: Typo

File: beacon-chain/core/types/state.go [hunk 3]
```diff
@@ -192,9 +224,60 @@ func (b *BeaconState) LatestCrosslinks() []*pb.CrosslinkRecord {
...
+}
+
+// LatestAttestations returns the latest pending attestations that have not been
+// processed.
+func (b *BeaconState) LatestAttestations() []*pb.PendingAttestationRecord {
+	return b.data.LatestAttestations
+}
+
+// ProcessedPowReceiptRootHash32 returns the root hashes of the
+// transaction receipts from the POW chain.
```
Comment (terenc3t) [inline beacon-chain/core/types/state.go]: s/transaction receipts/processed transaction receipts/

File: beacon-chain/core/types/state.go [hunk 4]
```diff
@@ -192,9 +224,60 @@ func (b *BeaconState) LatestCrosslinks() []*pb.CrosslinkRecord {
...
+	return b.data.ProcessedPowReceiptRootHash32
+}
+
+// CandidatePowReceiptRoots returns the root records of receipts that have
+// yet to be processed.
+func (b *BeaconState) CandidatePowReceiptRoots() []*pb.CandidatePoWReceiptRootRecord {
+	return b.data.CandidatePowReceiptRoots
+}
+
+// GenesisTime returns the creationg time of the
```
Comment (terenc3t) [inline beacon-chain/core/types/state.go]: typo *creationg

**Pull request merged**